app/presenters/notification.rb

```diff
@@ -0,0 +1,96 @@
+module ExercismWeb
+  module Presenters
+    class Notification
+      include ExercismWeb::Helpers::FuzzyTime
+
+      def self.prepare(item, current_user)
+        case item.regarding
+        when "code"
+          CodeNotification.new(item, current_user)
+        when "nitpick"
+          NitpickNotification.new(item, current_user)
+        when "like"
+          LikeNotification.new(item, current_user)
+        end
+      end
+
+      def initialize(item, current_user)
+        @notification = item
+        @current_user = current_user
+      end
+
+      def content_on_notification_page
+        if notification.item
+          html = "<li class='alert-row'>"
+          html += "<span style='padding-right: 5px;' class='fa #{icon_klass}'></span>"
+          html += large_content + " "
+          html += ago(notification.created_at)
+          html += "</li>"
+        end
+      end
+
+      def content_on_dashboard_page
+        if notification.item
+          html = "<li style='list-style-type: none; margin: 0;'>"
+          html += shortened_content
+          html += "</li>"
+        end
+      end
+
+      def large_content
+        raise NotImplementedError, "This #{self.class} cannot respond to large_content"
+      end
```

Comment:
The large content has the same format in each case:

```plain
$USER liked your submission of $PROBLEM in $LANGUAGE $AGO.
$USER submitted a new iteration of $PROBLEM in $LANGUAGE $AGO.
$USER commented on your submission of $PROBLEM in $LANGUAGE $AGO.
$USER commented on their submission of $PROBLEM in $LANGUAGE $AGO.
$USER commented on $CREATOR's submission of $PROBLEM in LANGUAGE $AGO.
```

If you give the notification the data necessary for the bits that change, then it will be easy to put the html in the erb file with no conditionals.